### PR TITLE
Setting PTPTC to None for Py3.9 only

### PR DESCRIPTION
**Pull request by vrdmr**
## Description
- Made the changes for Python 3.9 only.
- Added tests for verifying that the default is None only for Py3.9 and not for others.

---

### PR information

- [x] The title of the PR is clear and informative.
- [x] There are a small number of commits, each of which has an informative message. This means that previously merged commits do not appear in the history of the PR. For information on cleaning up the commits in your pull request, [see this page](https://github.com/Azure/azure-powershell/blob/master/documentation/development-docs/cleaning-up-commits.md).
- [ ] If applicable, the PR references the bug/issue that it fixes in the description.
- [x] New Unit tests were added for the changes made and CI is passing.

### Quality of Code and Contribution Guidelines
- [x] I have read the [contribution guidelines](https://github.com/Azure/azure-functions-python-worker/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

File: tests/unittests/test_dispatcher.py [hunk 1]
```diff
@@ -344,6 +297,13 @@ async def test_async_invocation_request_log_in_placeholder_threads(self):
                     'function type: async'
                 )
 
+    async def _assert_workers_threadpool(self, ctrl, host, expected_worker_count):
```
Comment (Hazhzeng) [inline tests/unittests/test_dispatcher.py:300]: Thanks for refactoring this

File: tests/unittests/test_dispatcher.py [hunk 2]
```diff
@@ -392,4 +352,39 @@ async def _check_if_async_function_is_ok(self, host) -> Tuple[str, str]:
...
+        self.mock_version_info.stop()
+
+
+class TestThreadPoolSettingsPython39(TestThreadPoolSettingsPython37):
+    def setUp(self):
+        super(TestThreadPoolSettingsPython39, self).setUp()
+
+        # 6 - based on 2 cores - min(32, (os.cpu_count() or 1) + 4) - 2 + 4
+        self._default_workers: Optional[int] = 6
+        self.mock_os_cpu = patch(
```
Comment (Hazhzeng) [inline tests/unittests/test_dispatcher.py:378]: We also want to start the patching mock_os_cpu here

Comment (vrdmr) [reply tests/unittests/test_dispatcher.py:378]: Ah.. :) I missed this and this failed.

You should never do changes at 2 AM. :)

**Comment by Hazhzeng** [comment]
Python 3.6 CI is failing, please check

**Review by Hazhzeng — APPROVED** [review]

**Pull request merged**